jdaviz/cli.py

```diff
@@ -85,15 +88,17 @@ def main(filename, layout='default', instrument=None, browser='default',
         os.chdir(start_dir)
 
 
-def _main():
+def _main(config=None):
     import argparse
     import sys
 
     parser = argparse.ArgumentParser(description='Start a Jdaviz application instance with data '
                                      'loaded from FILENAME.')
-    parser.add_argument('layout', choices=['cubeviz', 'specviz', 'specviz2d', 'mosviz', 'imviz'],
-                        help='Configuration to use.')
-    parser.add_argument('filename', type=str,
+    if config is None:
+        parser.add_argument('layout', choices=['cubeviz', 'specviz', 'specviz2d',
+                                               'mosviz', 'imviz'],
+                            help='Configuration to use.')
+    parser.add_argument('--filename', type=str, default=None,
```

Comment:
is this intentionally `--filename` now?  That seems to break passing filename as a positional argument.

Reply:
I don't think I changed those... This is on main: https://github.com/spacetelescope/jdaviz/blob/4366f754026d2ba3745e70b0a35634a413375668/jdaviz/cli.py#L96

Reply:
Oh wait, you're talking about the `--` part. Yes that's required for the change, as that's how `argparse` specifies an optional argument. Why isn't specifying a "default" keyword sufficient? I don't know...

Reply:
I don't think we need this at all. I think we can make the existing filename with nargs set to "no argument is acceptable".

https://docs.python.org/3/library/argparse.html#nargs

Reply:
Great find! Implemented in https://github.com/spacetelescope/jdaviz/pull/1890/commits/9fb79d0e2d7aa8542e1577a68f11b6ad91fb8105. For reference, i tried providing multiple filepaths expecting it to explode, but it actually returns a nice error message:

```
(envmain) PS E:\STScI\gitRepos\jdaviz> specviz ..\..\datasets\spec-0751-52251-0160.fits ..\..\datasets\spec-0751-52251-0160.fits
usage: specviz [-h] [--instrument INSTRUMENT] [--browser BROWSER] [--theme {light,dark}] [--verbosity {debug,info,warning,error}] [--history-verbosity {debug,info,warning,error}] [--hotreload | --no-hotreload]
               [--version]
               [filename]
specviz: error: unrecognized arguments: ..\..\datasets\spec-0751-52251-0160.fits
```

Reply:
@kecnry this should preserve the backwards compatibility and not require the `--filename` flag